evm/src/executor/fork/database.rs

```diff
@@ -114,8 +114,9 @@ impl ForkedDatabase {
     }
 
     pub fn revert_snapshot(&mut self, id: U256) -> bool {
-        let snapshot = { self.snapshots().lock().remove(id) };
+        let snapshot = { self.snapshots().lock().remove_at(id) };
         if let Some(snapshot) = snapshot {
+            self.snapshots().lock().insert_at(snapshot.clone(), id);
```

Comment:
Kind of looks like a deadlock from a glance but it's not haha—both guards are dropped on the spot